ephemeris_tools_unit_tests/ephem_tools_unit_tests.py

```diff
@@ -0,0 +1,702 @@
+            
+    for tool in tools:
+        tests_file = os.path.join(source_location[0], tool)
+        with open(tests_file, 'r') as test_urls:
+            test_urls = test_urls.readlines()
+    
+        if 'viewer3' in test_urls[0]:
+            viewer_replace(ephem, tests_file, golden_copies_location)
+        if 'tracker3' in test_urls[0]:
+            moon_tracker_replace(ephem, tests_file, golden_copies_location)
+        if 'ephem3' in test_urls[0]:
+            ephemeris_generator_replace(ephem, tests_file, 
+                                        golden_copies_location)
+            
+current_time = datetime.now().strftime('%H:%M:%S')
+logging.basicConfig(filename=f'ephem_tools_unit_test_{current_time}.log',
+                    encoding='utf-8',
+                    level=logging.DEBUG, 
+                    format='%(asctime)s | %(levelname)s | %(message)s',
+                    datefmt='%y-%m-%d %H:%M:%S',
+                    force=True)
+    
+warnings.simplefilter('ignore', urllib3.exceptions.InsecureRequestWarning)
+logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--run-test-ephemeris', dest='chosenephems', 
+                    action='append_const', const='test',
+                    help='Runs your chosen action with the "test" '
+                         'extension implemented to force all tests to operate '
+                         'with the most recent ephemeris.')
+
+parser.add_argument('--run-current-ephemeris', dest='chosenephems', 
+                    action='append_const', const='current',
+                    help='Runs your chosen action in the current active '
+                         'version of the server.')
+
+parser.add_argument('tool_directory', type=str, nargs=1,
+                    help='The location of the URL source files that will '
+                         'generate the results. All files called must be '
+                         'within the same location within the directory. '
+                         'Do not include the file name within the path. '
+                         'Do not end the path with a /.')
+
+parser.add_argument('golden_directory', type=str, nargs=1, 
+                    help='Path to the golden_files storage.')
+
+parser.add_argument('--comparison', action='store_true',
+                    help='Compares results of chosen ephemeris tools to '
+                         'their stored golden version counterparts. '
+                         'The generated results are pulled from your '
+                         'chosen server.')
+
+parser.add_argument('--replacement', dest='folderaction', action='store_false',
+                    help='Replaces stored versions of chosen ephemeris '
+                         'tools. All versions stored are generated from the '
+                         'current staging server.')
+
+parser.set_defaults(folderaction=True)
+
+parser.add_argument('--tools', type=str, nargs='+',
```

Comment:
I'm confused by this argument. Is it a list of tool names? Or a list of filenames?

Based on how its used, it appears to be a list of filenames.

But then why is there a separate option for the `tool_directory`? Why not just have people specify the entire filename including path here? Or if you want the options to not look like filenames (e.g. not have `.txt` extensions) then don't require an actual filename here, just ask for some shortened name.